cEP-0024.md

```diff
@@ -0,0 +1,252 @@
+plugins which support coala development workflow. So, this project
+adds plugins to auto-review, auto-reject, auto-label PRs, assign, unassign
+issues, squash commits.
+
+## Changes in coala workflow
+
+### Use increasing difficulty level order to assign issues
+
+The current assign process in coala is to check team membership of a user,
+difficulty level of the issue and assign issue to user if eligible. To
+implement the current assign process in GitMate, IGitt would need to support
+teams.
+
+Currently, IGitt doesn't support teams and there are no teams in GitLab.
+Considering [GitLab groups](https://docs.gitlab.com/ee/user/group/) and
+[GitHub teams](https://help.github.com/articles/about-teams/) as teams in
+IGitt is a possible solution for IGitt implementation. But, it requires extra
+tools to sync GitHub team and GitLab group memberships, which is a manual
+process and isn't automated yet.
+
+Also, team membership information is not publicly accessible in GitHub, so any
+system which needs to implement the current assign system would require
+organization authentication tokens.
+
+Best possible solution would be using an increasing difficulty level order,
+so that a user can be assigned to an issue only if user has solved issues of
+lower difficulty level before requesting to get assigned to a higher difficulty
+level issue.
+
+#### Drawbacks
+
+If user completed the lower difficulty level issues in GitHub and want to
+work on a higher difficulty level issue in GitLab, he/she has to repeat the
+newcomer process in GitLab. To resolve this issue, GitMate needs to find a way
+to link GitHub org and GitLab org, this would be easy when they have same name,
+as in the case of `coala`. But, resolving this issue is out of scope for this
+project.
+
+If a user is worthy enough to work on the higher level difficulty issues
+directly, then that user needs to be manually assigned by the maintainers until
+that user completes at least one low difficulty issue. Even with current assign
+system, manual intervention by maintainers is needed to add user to developers
+team.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened or updated, if any of PR's commit message referenced
+any issues with keywords `Fixes` or `Closes` and there is no
+CI failure in PR, then auto assign PR author to the mentioned unassigned
+issues.
+
+If someone else is working on the mentioned issue, comment will be added
+on the PR (configurable).
+
+If PR author is not a member of the organization, then he/she can't be assigned
+to any of the issues, a label(configurable) will be added on the PR so that
+maintainers can identify these type of PRs and a comment will be added on PR
+informing PR author to get invited to the organization before working on any
+issue.
+
+coala can configure this plugin as:
+
+**Comment message when someone else is assigned to mentioned issues(use
```

Comment:
again, no space before round bracket.

And like most places, here the use of the round bracket is where the communication with the reader breaks down.

The configuration item is "Comment message when someone else is assigned to mentioned issues", and should be bold

"(Use {issues}` as placeholder to get links of all assigned issues)" is a minor note for how to interpret the message, but it isnt part of the description of the configuration, and shouldnt be bold, and should be separate from the label.

Whitespace helps the reader pause, digest the first part before continuing to the next part.

Technically, grammatically `barry foo(bar baz)` means "(bar baz)" is a modifier/qualifier of some sort for the word "foo" *only*, and is not relevant for the word "barry".